Kudu.Agent/Controllers/KuduProcessController.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Kudu.Agent.Contracts;
+using Kudu.Agent.Util;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Authentication;
+using System.Threading;
+
+namespace Kudu.Agent.Controllers
+{
+    [ApiController]
+    [Route("/processes")]
+    public class KuduProcessController : ControllerBase
+    {
+        [HttpGet]
+        public IActionResult GetAllProcesses()
+        {
+            IEnumerable<ProcessInfo> processes = Process.GetProcesses()
+                .Where(p => !AppServicePlatformProcess(p))
+                .Select(p => GetProcessInfo(p, details: false));
+
+            return Ok(processes);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetProcess(int id)
+        {
+            Process p = GetProcessById(id);
+            if (p == null)
+            {
+                return NotFound();
+            }
+
+            ProcessInfo info = GetProcessInfo(p, details: true);
+
+            return Ok(info);
+        }
+
+        [HttpDelete, HttpPost]
+        public IActionResult KillProcess(int id)
+        {
+            Process p = GetProcessById(id);
+            p.Kill(entireProcessTree: true);
+
+            return Ok();
+        }
+
+        [HttpGet("{id}/threads")]
+        public IActionResult GetAllThreads(int id)
+        {
+            Process process = GetProcessById(id);
```

Comment:
When an invalid process id is passed Process.GetProcessById called from private GetProcessById would throw an exception, in that case this should return NotFound instead of throwing 500